daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1395,11 +1395,28 @@ private[lf] object SBuiltin {
       checkToken(token)
       args.get(0) match {
         case SOptional(opt) =>
-          opt match {
-            case None =>
-              unwindToHandler(machine, payload) //re-throw
-            case Some(handler) =>
-              machine.enterApplication(handler, Array(SEValue(token)))
+          machine.withOnLedger("SBTryHandler") { onLedger =>
```

Comment:
is there a reason why we cannot extend `OnLedgerBuiltin`?

Reply:
Yes, probably could here.